### PR TITLE
PR - Update assessment status as NC error

### LINKED ISSUE #1913: Update assessment status as NC error
To replicate this: try to change an Assessment status with a no-administrator user.

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1913

### CODE REVIEW

File: src/meta/user/authorizer.ts [hunk 1]
```diff
@@ -113,8 +113,53 @@ const canEdit = (props: {
...
+ * @param props.assessment
+ * @param props.countryIso
+ * @param props.cycle
+ * @param props.table
+ * @param props.status
+ * @param props.user
+ * @returns boolean
+ */
+const canEditAssessmentStatus = (props: {
+  countryIso: CountryIso
```
Comment (minotogna) [inline src/meta/user/authorizer.ts]: Herr countryiso can be removed and taken from country

File: src/meta/user/authorizer.ts [hunk 2]
```diff
@@ -113,8 +113,53 @@ const canEdit = (props: {
...
+
+  return false
+}
+
 export const Authorizer = {
   canView,
   canViewUsers,
-  canEdit,
+  canEditAssessmentStatus,
+  canEditSections,
```
Comment (minotogna) [inline src/meta/user/authorizer.ts]: Can this be rolled back to canEdit ?

Comment (rfontanarosa) [reply src/meta/user/authorizer.ts]: ok

File: src/meta/user/authorizer.ts (hunk 1 above)
Comment (rfontanarosa) [reply src/meta/user/authorizer.ts]: done

**Comment by minotogna** [comment]
Can you rename the Pr please?

File: src/meta/user/authorizer.ts [hunk 3]
```diff
@@ -113,8 +112,46 @@ const canEdit = (props: {
...
+    (Users.isNationalCorrespondent(user, countryIso, cycle) ||
+      Users.isAlternateNationalCorrespondent(user, countryIso, cycle)) &&
+    status === AssessmentStatus.editing
+  )
+    return true
+
+  if (Users.isReviewer(user, countryIso, cycle)) {
+    return [AssessmentStatus.editing, AssessmentStatus.review].includes(status)
+  }
+
```
Comment (minotogna) [inline src/meta/user/authorizer.ts]: Don't we have A centralized place
To check status allowance?

File: src/server/api/area/index.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { postCountry } from './postCountry'
 export const AreaApi = {
   init: (express: Express): void => {
     // Country
-    express.post(ApiEndPoint.Area.country(), AuthMiddleware.requireEditTableData, postCountry)
+    express.post(ApiEndPoint.Area.country(), AuthMiddleware.requireEditAssessmentStatus, postCountry)
```
Comment (minotogna) [inline src/server/api/area/index.ts]: This end point should also be specific to edit assessment status.
It should be something like countrySstatus

Comment (rfontanarosa) [reply src/server/api/area/index.ts]: that endpoint allows to edit all the country properties, not only the status, so probably requireEditAssessmentStatus should be renamed into requireEditCountryAssessment

File: src/meta/user/authorizer.ts (hunk 3 above)
Comment (rfontanarosa) [reply src/meta/user/authorizer.ts]: nope, we don't :( we have getRolesAllowedToEdit but it's not related to status

**Review by sorja — APPROVED** [review]

File: src/meta/user/authorizer.ts (hunk 3 above)
Comment (minotogna) [reply src/meta/user/authorizer.ts]: Can't we get status allowed to edit and then check if the status passed to the middleware is included in those ?

File: src/server/api/area/index.ts (hunk 4 above)
Comment (minotogna) [reply src/server/api/area/index.ts]: canEditCountryProps and requireEditCountryProps ?
What do you think ?

File: src/meta/user/authorizer.ts (hunk 3 above)
Comment (rfontanarosa) [reply src/meta/user/authorizer.ts]: mmmm, can we do that in another pr? at the moment the status check is used by canEditData and canEditCountryProps and the logic is "check the role then check the status"
we have to reverse that logic "get the roles allowed by the status then check the role"

**Review by minotogna — APPROVED** [review]

File: src/meta/user/authorizer.ts (hunk 3 above)
Comment (minotogna) [reply src/meta/user/authorizer.ts]: sure, let's talk in the next daily

**Pull request merged**